getting-started/app-services.html.md

```diff
@@ -0,0 +1,154 @@
+
+#### Private (Flycast) apps
+
+To have Fly Proxy load-balance among VMs on a non-public app, [allocate a _private_ (Flycast) IPv6 address](/docs/reference/private-networking/#assigning-a-flycast-address) using `fly ips allocate-v6 --private`, and **remove any public IPs from the app** using `fly ips release`. A Flycast IP can only be reached from within the private network it's allocated on, which must belong to the same Fly Organization as the Fly App it points to. 
+
+<section class="warning icon">
+If your configuration includes any services for Fly Proxy to route to, and the app has a public IP, that service is exposed to the whole internet.</section>
+
+### A note on Fly App IPs and Fly-Replay routing
+
+App-wide IP addresses (public and private) tell Fly Proxy which app to deliver a request to. If a service _only_ needs to be reachable by Fly Proxy in its handling of the [Fly-Replay](/docs/reference/dynamic-request-routing/) response header, its app does not need an IP; app information is incorporated directly in the header.
+
+## Private services on 6PN (direct WireGuard connections)
+
+If you don't need load balancing or other Fly Proxy features, you (or one of your apps) can connect directly on a VM's 6PN address: its address on its IPv6 private WireGuard network.
+
+<section class="callout">By default, all of an organization's Fly Apps live on the same private network, but you can create apps on different private networks to isolate them from one another. You'd want to do this if you run separate apps for your customers, for example.</section>
+
+### Get the service listening on the right address within the VM
+
+Each VM has the hostname `fly-local-6pn` mapped to its own 6PN address in its `/etc/hosts` file, so to make a service reachable over the private network, bind it to `fly-local-6pn:<port>`. Binding to a port on the IPv6 wildcard "address" (`[::]:<port>`) also works (but see [A note on IPv4 and IPv6 wildcards](#a-note-on-ipv4-and-ipv6-wildcards)). 
+
+That's it, really. There's no need to define any service in `fly.toml`, and there's no need to provision any IPs for the app.
+
+If your service works publicly but can't be reached directly over WireGuard, the problem may be that it's not listening on IPv6.
+
+### Fly.io internal DNS and 6PN addresses 
+
+Fly.io runs a specialized internal DNS service to find 6PN addresses for VMs. Queries on the various `.internal` names return either AAAA records for 6PN addresses, or a TXT record with information about your app, your network, or your Machines. 
+
+Internal DNS gives us some useful presets for targeting requests. For instance, a DNS query on the address `<region>.<appname>.internal` yields the 6PN addresses of the VMs of app `<appname>` in region `<region>`. Addressing a request to `<region>.<appname>.internal` gets it delivered to one of those VMs.
```

Comment:
```suggestion
Internal DNS can give you some useful presets for targeting requests. For instance, a DNS query on the address `<region>.<appname>.internal` yields the 6PN addresses of the VMs of app `<appname>` in region `<region>`. Addressing a request to `<region>.<appname>.internal` gets it delivered to one of those VMs.
```

Reply:
Disagree. 😄 The presets _do_ exist; we _can_ use them.